examples/rogue_associations.rb

```diff
@@ -0,0 +1,96 @@
+# This attempts to illustrate the redis way of...
+#   - has_one 1:1 (Rogue - equipment)
+#   - has_many through (Rogue - inventory) 1:1 many:many 1:1
+#   - has_one 1:many (equipment - item)
+#   - Mixin usage to promote DRY code
+
+require "ohm"
+
+def battleable
+  attribute :name;       index :name
+  attribute :hp,         lambda { |x| x.to_i } # cast hp to integer
+  attribute :mp,         lambda { |x| x.to_i }
+  list      :inventory,  :Item
+
+  attribute :equipment_id
+  reference :equipment,  :Loadout
+end
+
+# We define both a `Video` and `Audio` model, with a `list` of *comments*.
+class Rogue < Ohm::Model
+  battleable # give Rogues all the attributes required to conduct battle
+
+  # instead of using Rogue.create, favor Rogue.spawn
+  def self.spawn(name: nil)
+    rogue = self.create(name: name)
+    rogue.equipment = Loadout.create
+    rogue.save
+    rogue
+  end
+end
+
+class Monster < Ohm::Model
+  battleable # give monsters attributes required to conduct battle
+
+
+end
+
+class Loadout < Ohm::Model
+  attribute :head_id
+  reference :head,       :Item
```

Comment:
The `reference` macro already defines the `head_id` attribute, so all the `attribute :*_id` lines can be removed.